lib/puppet/provider/cisco_evpn_multicast/cisco.rb

```diff
@@ -0,0 +1,69 @@
+# February, 2018
+#
+# Copyright (c) 2018-2019 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'cisco_node_utils' if Puppet.features.cisco_node_utils?
+
+Puppet::Type.type(:cisco_evpn_multicast).provide(:cisco) do
+  desc 'The Cisco provider for cisco evpn multicast'
+
+  confine feature: :cisco_node_utils
+
+  mk_resource_methods
+
+  def initialize(value={})
+    super(value)
+    @property_flush = {}
+  end # initialize
+
+  def self.properties_get
+    new(
+      ensure: :present,
+      name:   'default')
+  end # self.properties_get
+
+  def self.instances
+    inst = []
+    evpn_multicast = Cisco::EvpnMulticast.multicast
+    inst << properties_get unless evpn_multicast == ''
+    inst
+  end # self.instances
+
+  def self.prefetch(resources)
+    resources.values.first.provider = instances.first unless instances.first.nil?
+  end
+
+  def exists?
+    @property_hash[:ensure] == :present
+  end
+
+  def create
+    @property_flush[:ensure] = :present
+  end
+
+  def destroy
+    @property_flush[:ensure] = :absent
+  end
+
+  def flush
+    if @property_flush[:ensure] == :absent
+      @nu = Cisco::EvpnMulticast.new
```

Comment:
I believe you can avoid creating the object here just to destroy it by moving the call `@nu = Cisco::EvpnMulticast.new` into the initialize method.

Example: https://github.com/cisco/cisco-network-puppet-module/blob/develop/lib/puppet/provider/cisco_bfd_global/cisco.rb#L65

Reply:
@mikewiebe .. I removed that because of the way I've written the NU piece for this. I call create (in node utils) every single time and this was causing it causing evpn_multicast to be turned on while doing 'puppet resource'. Since this is really an ensure present/absent kind of provider I moved the nu object creation to flush.

Reply:
Gotcha.  Maybe add a clarification comment.

Reply:
Done.